typings/styled-components.d.ts

```diff
@@ -12,7 +12,7 @@ export type StyledProps<P> = ThemedStyledProps<P, any>;
 
 export type ThemedOuterStyledProps<P, T> = P & {
   theme?: T;
-  innerRef?: (instance: any) => void;
+  innerRef?: ((instance: any) => void) | RefObject<HTMLElement | SVGElement>
```

Comment:
Since styled-components can style any component that has `className` property, it may happen that ref would not be only a dom element.
The previous callback-based ref assumed that by `any`. The new one can be `object`, in my opinion.

Reply:
It's a React component instance or HTML DOM element - so `object` should be fine. I'm not sure if there's anything in the `react` or `react-dom` typings that is even more specific.

For now, i'll update it to object.